Python_DataEngineering/Extract_Store_Fetch.py

```diff
@@ -0,0 +1,158 @@
+                if IST not in timing_snap_shot:
+                    timing_snap_shot[IST]=int(properties["value"])
+                else:
+                    timing_snap_shot[IST]+=int(properties["value"])
+    return timing_snap_shot
+
+
+#This function inserts all the required values from API into twoo table
+#using sql1 table we will insert bikepointname, no of available bikes, datetime
+#using sql2 table we will insert total no of bikes present at a particular timesnapshot
+def databaseInsert(cursor,db,time_snap_shot_bike_data,bike_point_details):
+    sql1="insert into Bike_Detail(bikepointname, noofbike, datetime) VALUES (%s, %s, %s)"
+
+    #inserting into Total_Bike_Table using sql2
+    sql2="insert into time_snap_shot_detail(datetime, noofbike) VALUES (%s, %s)"
+    for date_time in time_snap_shot_bike_data:
+        datetime=date_time
+        no_of_bikes=str(time_snap_shot_bike_data[date_time])
+        value=(datetime,no_of_bikes)
+        cursor.execute(sql2,value)
+        db.commit()
+        
+    
+    
+    #inserting into Bike Table using sql1
+    for bikepoint in bike_point_details:
+        bikepointname=bikepoint
+        noofbikes=bike_point_details[bikepoint][0]
+        datetime=bike_point_details[bikepoint][1]
+        value=(bikepointname,noofbikes,datetime)
+        cursor.execute(sql1,value)
+        db.commit()
+        
+    
+     
+#this function will fetch no of bikes available as per user's selection of bikepointname and currenttime
+def databaseFetch(cursor,db,select,datetime):
+    sql="select noofbike from Bike_Detail where bikepointname= %s and datetime= %s"
+    value=(select,datetime,)
+    cursor.execute(sql,value)
+    result=cursor.fetchall()
+    if cursor.rowcount == 0:
+        print("No Bikes available at this point of Time")
+    else:
+        for each_row in result:
+            print(select+" : "+each_row[0])
+            
+ #creates the required table           
+def createTables(cursor,db):
+    sql1="create table Bike_Detail(bikepointname varchar(255),noofbike varchar(255),datetime varchar(255))"
+    sql2="create table time_snap_shot_detail(datetime varchar(255),noofbike varchar(255))"
+    cursor.execute(sql1)
+    db.commit()
+    cursor.execute(sql2)
+    db.commit()
+    
+    
+        
+
+#This executes as a main function where the subfunction are called
+def modules(cursor,db):
+    bikepointdetails=bikesPerBikePoint(json_data)
+    time_snap_shot_bike_data=bikesPerTimeSnapShot(json_data)
+    databaseInsert(cursor,db,time_snap_shot_bike_data,bikepointdetails)
+
+
+    
+if __name__ == "__main__":
+    request=requests.get("https://api.tfl.gov.uk/bikepoint") #here i am extracting the API using get method
+    json_data=request.json() #converting the request object to JSON so as to extract the relevant Data
+    tuples=databaseConfig()
+    cursor=tuples[0]
+    db=tuples[1]
+    createTables(cursor,db)

```

Comment:
Creating tables on every execution might lead to an error if I'm executing this for the second time, right?

Reply:
No, here yhe function create Tables will be executed only once. And the function names modules will be executed every 5 minutes , where it will do all the task like Fetch reuired data and insert it into Tables.